packages/calendar-input/src/date-input.tsx

```diff
@@ -0,0 +1,275 @@
+        }
+      })
+    }
+  }
+
+  const focusNextPart = () => {
+    if (activeDatePart) {
+      const activeDatePartIndex = CalendarDate.format.indexOf(activeDatePart)
+      if (activeDatePartIndex + 1 < CalendarDate.format.length) {
+        setActiveDatePart(CalendarDate.format[activeDatePartIndex + 1])
+        setCursorPosition(
+          CalendarDate.getPartStartIndex(
+            CalendarDate.format[activeDatePartIndex + 1]
+          )
+        )
+      }
+    }
+  }
+  const focusPrevPart = () => {
+    if (activeDatePart) {
+      const activeDatePartIndex = CalendarDate.format.indexOf(activeDatePart)
+      if (activeDatePartIndex > 0) {
+        setActiveDatePart(CalendarDate.format[activeDatePartIndex - 1])
+        setCursorPosition(
+          CalendarDate.getPartStartIndex(
+            CalendarDate.format[activeDatePartIndex - 1]
+          )
+        )
+      }
+    }
+  }
+
+  const isStartCurrentPart = () => {
+    if (activeDatePart) {
+      return CalendarDate.getPartStartIndex(activeDatePart) === cursorPosition
+    }
+    return false
+  }
+  const isEndCurrentPart = () => {
+    if (activeDatePart) {
+      return CalendarDate.getPartEndIndex(activeDatePart) - 1 === cursorPosition
+    }
+    return false
+  }
+
+  const setPartValue = (part: DateTimePartsType, value) => {
+    let newValue = Number(value)
+    if (isNaN(newValue)) {
+      changeInputValue({...inputValue, [part]: value})
+    } else {
+      if (part === 'month' && newValue > 12) {
+        newValue = 12
+      } else if (part === 'day' && newValue > 31) {
+        newValue = 31
+      } else if (part === 'hours' && newValue > 23) {
+        newValue = 23
+      } else if (part === 'minutes' && newValue > 59) {
+        newValue = 59
+      }
+      changeInputValue({...inputValue, [part]: newValue})
+    }
+  }
+
+  const moveCursorRight = (count: number = 1) => {
+    for (let i = 0; i < count; i++) {
+      if (isEndCurrentPart()) {
+        focusNextPart()
+      } else if (
+        cursorPosition + 1 <
+        CalendarDate.getStringValueForParts(inputValue).length
+      ) {
+        setCursorPosition(cursorPosition + 1)
+      }
+    }
+  }
+
+  const moveCursorLeft = (count: number = 1) => {
+    for (let i = 0; i < count; i++) {
+      if (isStartCurrentPart()) {
+        focusPrevPart()
+      } else if (cursorPosition > 0) {
+        setCursorPosition(cursorPosition - 1)
+      }
+    }
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent<{}>) => {
+    e.stopPropagation()
+    if (e.keyCode === KEYS.DOWN) {
+      openCalendar()
+    } else if (e.keyCode === KEYS.LEFT) {
+      moveCursorLeft()
+    } else if (e.keyCode === KEYS.RIGHT) {
+      moveCursorRight()
+    } else if (e.keyCode === KEYS.HOME) {
+      setCursorPosition(0)
+      setActiveDatePart(CalendarDate.format[0])
+    } else if (e.keyCode === KEYS.END) {
+      const endPart = CalendarDate.format[CalendarDate.format.length - 1]
```

Comment:
It is possible to:
1) Select "day" part
2) Click "END" key
3) After input, it ends up in putting value >after< minutes:

![Zrzut ekranu 2019-12-3 o 09 04 59](https://user-images.githubusercontent.com/8265332/70031645-fefff680-15ab-11ea-8524-727e851b4eb5.png)